financial-templates-lib/ExpiringMultiPartyClient.js

```diff
@@ -1,9 +1,59 @@
+const { delay } = require("./delay");
+
+const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+
 // A thick client for getting information about an ExpiringMultiParty.
-export default class ExpiringMultiPartyClient {
-  constructor() {
+class ExpiringMultiPartyClient {
+  constructor(empAddress) {
     this.positions = [];
+    this.sponsorAddresses = [];
+    this.emp = new web3.eth.Contract(ExpiringMultiParty.abi, empAddress);
```

Comment:
Just curious why you chose to initialize a web3 instance rather than getting the truffle one that's deployed (or using `.at()` to create one from an address)?

Reply:
In order to get the events? Or are there other/better ways to do that?

Reply:
Oh, the web3 object you created exists inside the truffle contract object:
```
let emp = await ExpiringMultiParty.at(empAddress);
emp.contract.getPastEvents(...)
```
However, this constructor isn't async (so it might be difficult to initialize the truffle version), and there's nothing below that's any more complex using the web3 one, so I don't necessarily think it's better to use the truffle one.